tests/shim_test.go

```
@@ -0,0 +1,272 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/clearcontainers/shim/mock"
+	hypermock "github.com/containers/virtcontainers/hyperstart/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type testRig struct {
+	t *testing.T
+
+	// wait for the shim to finish
+	wg sync.WaitGroup
+
+	//Mock Proxy
+	proxy *mock.MockProxy
+
+	//shim forked
+	shimCommand *exec.Cmd
+
+	shimStdin  io.WriteCloser
+	shimStdout io.ReadCloser
+	shimStderr io.ReadCloser
+}
+
+func newTestRig(t *testing.T) *testRig {
+	return &testRig{
+		t: t,
+	}
+}
+
+const testContainerid = "123456789"
+const testToken = "testtoken"
+
+var shimPath string
+
+func TestMain(m *testing.M) {
+	flag.StringVar(&shimPath, "shimpath", "/usr/libexec/cc-shim",
+		"absolute path to the cc-shim executable")
+	flag.Parse()
+	fmt.Printf("Path of cc-shim executable : %s\n", shimPath)
+	os.Exit(m.Run())
+}
+
+func (rig *testRig) Start() {
+	proxySocketPath := hypermock.GetTmpPath("test-proxy.%s.sock")
+	rig.proxy = mock.NewMockProxy(rig.t, proxySocketPath)
+	rig.proxy.Start()
+
+	//Fork the shim
+	uri := "unix://" + proxySocketPath
+	rig.t.Logf("Uri passed to shim :%s\n", uri)
+
+	rig.shimCommand = rig.getShimCommand(uri)
+	err := rig.shimCommand.Start()
+	assert.Nil(rig.t, err)
+	rig.t.Logf("shim started\n")
+}
+
+func (rig *testRig) getShimCommand(uri string) *exec.Cmd {
```

Don't you think you could return the errors returned by StdinPipe(), StdoutPipe() and StderrPipe() ?

---

Its sufficient to assert here, there is nothing meaningful to be done with the return values. We cannot continue if this fails.